galaxy_ng/app/tasks/synclist.py

```diff
@@ -0,0 +1,147 @@
+    current_task = Task.current()
+    current_task.task_group = task_group
+    current_task.save()
+
+    GroupProgressReport(
+        message="Synclists curating upstream repo",
+        code="synclist.curate",
+        total=synclist_qs.count(),
+        task_group=task_group).save()
+
+    with ProgressReport(message="Synclists curating upstream repo task",
+                        code="synclist.curate.log",
+                        total=synclist_qs.count()) as task_progress_report:
+
+        for synclist in synclist_qs:
+            # TODO: filter down to just synclists that have a synclist repo
+            # locks need to be Model or str not int
+            locks = [str(synclist.id)]
+
+            task_args = (synclist.id,)
+            task_kwargs = {}
+
+            subtask = enqueue_with_reservation(curate_synclist_repository,
+                                               locks,
+                                               args=task_args,
+                                               kwargs=task_kwargs,
+                                               task_group=task_group,
+                                               )
+            task_progress_report.increment()
+            log.debug('subtask: %s', subtask)
+
+            progress_report = task_group.group_progress_reports.filter(code='synclist.curate')
+            log.debug('progress_report: %s', progress_report)
+
+            progress_report.update(done=F('done') + 1)
+
+    log.debug('progress_report: %s', progress_report)
+    log.debug("Finishing curating %s synclist repos based on %s update",
+              synclist_qs.count(), upstream_repository)
+
+    task_group.finish()
+    log.debug('Big Task finished after %s seconds', time.time() - task_start)
+
+
+def curate_synclist_repository(synclist_pk, **kwargs):
+    """Update a synclist repo based on it's policy and spec.
+
+    Update a curated synclist repo to use the latest versions from
+    upstream as specified by the synclist's policy and it's collections
+    and namespaces fields.
+
+    This is intended to work on one synclist and synclist repo at a time.
+    """
+
+    synclist = models.SyncList.objects.get(pk=synclist_pk)
+
+    upstream_repository = synclist.upstream_repository
+    latest_upstream = upstream_repository.latest_version()
+
+    log.info(
+        'Applying synclist "%s" with policy=%s to curate repo "%s" from upstream repo "%s"',
+        synclist.name,
+        synclist.policy,
+        synclist.repository.name,
+        upstream_repository.name,
+    )
+
+    locks = [synclist.repository]
+
+    namespaces = synclist.namespaces.filter().values_list('name', flat=True)
+
+    collection_versions = \
+        CollectionVersion.objects.filter(repositories=synclist.upstream_repository,
+                                         collection__namespace__in=namespaces,
+                                         is_highest=True) \
```

Comment:
If there are 2 new `CollectionVersion`s for a `Collection`, would we want to sync both and not just the highest version?

Not sure how `is_highest` is intended to be used here... since it is set by `pulp_ansible`, but at the `CollectionVersion` level, it won't indicate the highest version for a particular `Repository`... it might be as useful as earlier for `galaxy_ng` (for the _ui collections endpoint based on repo, we had to stop using it)